app/models/concerns/course/levels_concern.rb

```diff
@@ -41,4 +41,35 @@ def compute_level_number(experience_points)
     level = compute_level(experience_points)
     level ? level.level_number : 0
   end
+
+  # Returns the current level's experience point (exp) threshold for
+  # for a given number of experience_points.
```

Comment:
It is possible to get the current course user's level, ie. `current_course_user.level.threshold`. For the `.next`, there is one issue: the current methods handle some edge cases based on the level model implementation. 

To explain, given the following Course and levels: **Level 1 - 100, Level 2 - 300, Level 3 - 500**
It is possible for a CourseUser to have a `nil` `level` (if exp < 100), or a `nil` `level.next` (if exp > 500). If I do it as you suggested, I think I might need to do some hacks which might not be as nice. 

There are a few ways to solve this: 
1) Retain current design, 
2) Use your idea of `current_course_user.level` and `current_course_user.level.next`, or
3) We can try `current_course_user.current_level` (this method exists as `current_course_user.course.compute_level(exp)`) and `current_course_user.next_level`

What do you think?